webapp/lib/model.g.dart

```diff
@@ -411,11 +411,49 @@ void listenForUpdates<T>(
     querySnapshot.docChanges().forEach((documentChange) {
       var doc = documentChange.doc;
       log.verbose('Processing ${doc.id}');
-      changes.add(createModel(doc));
+      changes.add(createModel(DocSnapshot(doc.id, doc.data())));
     });
     listener(changes);
   });
 }
 
+/// A snapshot of a document's id and data at a particular moment in time.
+class DocSnapshot {
+  final String id;
+  final Map<String, dynamic> data;
+
+  DocSnapshot(this.id, this.data);
+}
+
+/// A batch update, used to perform multiple writes as a single atomic unit.
```

Comment:
Comments as elsewhere about these being a batch not an atomic unit.

Reply:
I think that as of the latest update, they are now atomic.